src/views/forms/AccountManagementForms/SourceSelectMenu.tsx

```diff
@@ -14,43 +17,73 @@ import { getTransferInputs } from '@/state/inputsSelectors';
 
 type ElementProps = {
   label?: string;
+  selectedExchange?: string;
   selectedChain?: string;
-  onSelectChain: (chain: string) => void;
+  onSelectChain: (name: string, type: 'chain' | 'exchange') => void;
 };
 
-export const ChainSelectMenu = ({ label, selectedChain, onSelectChain }: ElementProps) => {
+export const SourceSelectMenu = ({
+  label,
+  selectedExchange,
+  selectedChain,
+  onSelectChain,
+}: ElementProps) => {
   const stringGetter = useStringGetter();
   const { type, depositOptions, withdrawalOptions, resources } =
     useSelector(getTransferInputs, shallowEqual) || {};
   const chains =
     (type === TransferType.deposit ? depositOptions : withdrawalOptions)?.chains?.toArray() || [];
 
+  const exchanges =
+    (type === TransferType.deposit ? depositOptions : withdrawalOptions)?.exchanges?.toArray() ||
+    [];
+
   const chainItems = Object.values(chains).map((chain) => ({
     value: chain.type,
     label: chain.stringKey,
     onSelect: () => {
-      onSelectChain(chain.type);
+      onSelectChain(chain.type, 'chain');
     },
     slotBefore: <Styled.Img src={chain.iconUrl} alt="" />,
   }));
 
-  const selectedOption = chains.find((item) => item.type === selectedChain);
+  const exchangeItems = Object.values(exchanges).map((exchange) => ({
+    value: exchange.type,
+    label: exchange.string,
+    onSelect: () => {
+      onSelectChain(exchange.type, 'exchange');
+    },
+    slotBefore: <Styled.Img src={exchange.iconUrl} alt="" />,
+  }));
+
+  const selectedChainOption = chains.find((item) => item.type === selectedChain);
+  const selectedExchangeOption = exchanges.find((item) => item.type === selectedExchange);
 
   return (
     <SearchSelectMenu
       items={[
-        {
+        exchangeItems.length > 0 && testFlags.showCEXDepositOption && {
+          group: 'exchanges',
+          groupLabel: stringGetter({ key: STRING_KEYS.EXCHANGES }),
+          items: exchangeItems,
+        },
+        chainItems.length > 0 && {
           group: 'chains',
           groupLabel: stringGetter({ key: STRING_KEYS.CHAINS }),
           items: chainItems,
         },
-      ]}
+      ].filter(isTruthy)}
       label={label || (type === TransferType.deposit ? 'Source' : 'Destination')}
     >
       <Styled.ChainRow>
-        {selectedChain ? (
+        {selectedChainOption ? (
+          <>
+            <Styled.Img src={selectedChainOption.iconUrl} alt="" /> {selectedChainOption.stringKey}
+          </>
+        ) : selectedExchangeOption ? (
           <>
-            <Styled.Img src={selectedOption?.iconUrl} alt="" /> {selectedOption?.stringKey}
+            <Styled.Img src={selectedExchangeOption.iconUrl} alt="" />{' '}
+            {selectedExchangeOption.stringKey}
```

Comment:
?: `stringKey` used outside of `stringGetter`

Reply:
Should be string, not sure why I had it as stringKey before 😅